File: HelloWorldController.java
package com.example;

import io.micronaut.context.LocalizedMessageSource;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;

import java.util.Optional;

@Controller
public class HelloWorldController {

    private final LocalizedMessageSource messageSource;

    public HelloWorldController(LocalizedMessageSource messageSource) {
        this.messageSource = messageSource;
    }

    @Produces(MediaType.TEXT_PLAIN)
    @Get
    Optional<String> index() {
        return messageSource.getMessage("hello.world");
    }
}

File: guides/localized-message-source/python/src/example/micronaut/hello_world_controller.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import LocalizedMessageSource
from micronaut.http import MediaType
from micronaut.http.annotation import Get

message_source: Annotated[LocalizedMessageSource, Inject]  # <1>


# tag::clazz[]
@Get(value="/", produces=MediaType.TEXT_PLAIN)  # <2> <3>
def index() -> str:
    return message_source.getMessage("hello.world").orElse("Hello World")  # <4>
# end::clazz[]
